clustered/common/src/main/java/org/ehcache/clustered/common/ServerStoreConfiguration.java

```diff
@@ -159,6 +159,13 @@ public String getResourcePoolName() {
         return resourcePoolName;
       }
     }
+
+    /**
+     * Creates a Pool Allocation which inherits the Shared or Fixed Pool Allocation from a cache which is already configured on the server.
+     */
+    final class Thin implements PoolAllocation {
```

Comment:
This class (and the other inner classes here) should probably all be static, right?